### PR TITLE
Remove team-member requirement for budget-line status changes

### PR DESCRIPTION
**Pull request by jonnalley**
## What changed

Stakeholders decided that submitting a status change (`ChangeRequest`) for one or more budget lines on an Agreement (e.g. Draft → Planned) should **not** require team members to be added to the agreement. Having a COR (`project_officer_id`) is sufficient — and that requirement was already enforced.

The team-member requirement was implemented **entirely as frontend Vest validation** (the backend already only requires the COR for a status change, and no backend test asserts team members). This PR removes that frontend requirement and updates the tests to prove the COR-only path. No backend or DB change was needed.

**`frontend/src/pages/agreements/review/suite.js`** — Deleted the `team-members` completeness test that gated the "Send to Approval" button on the Review page.

**`frontend/src/components/Agreements/AgreementEditor/AgreementEditFormSuite.js`** — Deleted the `team-members` completeness test (fired in review mode).

**`frontend/src/components/Agreements/AgreementEditor/AgreementEditForm.hooks.js`** — Removed the now-dead `runValidate("team-members", ...)` calls in the `setSelectedTeamMembers` / `removeTeamMember` handlers (they referenced the deleted Vest test). Adding/removing team members still dispatches correctly.

**`frontend/src/helpers/utils.js`** — Removed the orphaned `"team-members": "Team Members"` entry from the `validation` display-label map. Its only consumer was the deleted error key (rendered via `convertCodeForDisplay("validation", key)` in `ReviewAgreement.jsx`).

**`frontend/src/pages/agreements/review/suite.test.js`** — Removed the `team-members` assertion from the "missing required fields" test, and converted "fails if team_members is empty" into a positive "passes if team_members is empty" test.

**`frontend/cypress/e2e/createAgreementWithValidations.cy.js`** — Removed `team_members` from the API fixture and the team-member UI step, and added a positive API assertion (COR present, zero team members) to the Draft → Planned send-to-approval test, proving COR-only submission.

**`.claude/skills/`** — Added two reusable skills surfaced by this change: `dead-code-after-removal-sweep` (finding artifacts orphaned by a deletion) and `vest-error-key-lifecycle` (the producer→consumer chain a Vest error key travels).

_Explicitly out of scope:_ the "Only team members on this agreement can send to approval" authorization tooltips are a separate authorization check (`check_user_association()`, which already treats the COR as authorized). The Team Members UI field stays optional with no styling changes.

## Issue

https://github.com/HHS/OPRE-OPS/issues/5882

## How to test

**Unit tests:**
```bash
cd frontend
bun run test --watch=false src/pages/agreements/review/suite.test.js
bun run test --watch=false src/components/Agreements/AgreementEditor/
```
Expect all to pass, including the new "passes if team_members is empty" test.

**Lint/format:**
```bash
cd frontend && bun run format && bun run lint
```

**Cypress E2E** (Docker stack running):
```bash
cd frontend && bun run test:e2e
```
Confirm `createAgreementWithValidations.cy.js` passes — specifically "enables review status transitions with a COR but no team members".

**Manual (Docker stack up):**
1. Open an agreement that has a COR assigned but **no** team members, with at least one Draft budget line.
2. Go to Review / request a Draft → Planned status change; select the Draft budget line(s).
3. Confirm there is no "Team Members … required to submit for approval" error and the **Send to Approval** button is enabled (assuming all other required fields + the COR are present).

## A11y impact

- [ ] No accessibility-impacting changes in this PR
- [ ] Accessibility changes included and validated against WCAG 2.1 AA intent
- [ ] Any temporary suppression includes `A11Y-SUPPRESSION` metadata (owner, expires, rationale)

## Storybook

- [ ] No UI component changes in this PR
- [ ] Story added for new component in `src/components/UI/`
- [ ] Story updated to reflect changed props/states
- [x] N/A — change is page-specific or non-visual

## Screenshots

N/A — this change removes a validation requirement; no visual changes.

## Definition of Done Checklist
- [x] OESA: Code refactored for clarity
- [x] OESA: Dependency rules followed
- [x] Automated unit tests updated and passed
- [ ] Automated integration tests updated and passed
- [ ] Automated quality tests updated and passed
- [ ] Automated load tests updated and passed
- [x] Automated a11y tests updated and passed
- [x] Automated security tests updated and passed
- [ ] 90%+ Code coverage achieved
- [x] Form validations updated

## Links

N/A

### CODE REVIEW

File: frontend/src/components/Agreements/AgreementEditor/AgreementEditFormSuite.js [hunk 1]
```diff
@@ -53,9 +53,6 @@ const suite = create((data = {}, fieldName) => {
         enforce(data.contract_type).notEquals("-Select an option-");
         enforce(data.contract_type).isNotEmpty();
     });
```
Comment (weimiao67) [inline frontend/src/components/Agreements/AgreementEditor/AgreementEditFormSuite.js:55]: frontend/src/components/Agreements/AgreementEditor/AgreementEditForm.jsx:597 — orphaned error consumer left behind after its producer was deleted - the form still reads the team-members error key, but the AgreementEditFormSuite deleted team-members test().

**Review by weimiao67 — APPROVED** [review]

**Review by rajohnson90 — APPROVED** [review]
